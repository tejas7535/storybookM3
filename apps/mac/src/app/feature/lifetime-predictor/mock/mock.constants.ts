/* eslint-disable max-lines */
import { EChartsOption } from 'echarts';

import {
  BurdeningType,
  Material,
  Prediction,
  PredictionResult,
} from '../models';
import { LTPState } from '../store';
import { initialState } from '../store/reducers/input.reducer';
import { PredictionState } from '../store/reducers/prediction.reducer';

export const mockedMaterials: Material[] = [
  { name: 'Plastik', heatTreatment: 'volleHitze', hardness: 90_001 },
  { name: 'Holz', heatTreatment: 'halbeHitze', hardness: 1 },
];

export const mockedPredictions: Prediction[] = [
  { id: 0, name: 'Arnold Strong' },
  { id: 1, name: 'Silvester Stallone' },
];

export const mockedBurdeningTypes: BurdeningType[] = [
  { id: 0, name: 'Jet Lee' },
  { id: 1, name: 'Tyson Fury' },
];

export const mockedLoadsResult = {
  loads: {
    x: [96, 214],
    y: [1050, 1000],
  },
  woehler: {
    case: 1,
    fatigue_strength: 129.603_510_582_840_22,
    line_percentile_1: [
      [
        [1_158_691.416_562_977_7, 129.603_510_582_840_27],
        [10_000, 301.211_650_558_652_76],
      ],
      [
        [1_158_691.416_562_977_7, 129.603_510_582_840_27],
        [10_000_000, 129.603_510_582_840_27],
      ],
    ],
    line_percentile_10: [
      [
        [1_158_691.416_562_977_7, 129.603_510_582_840_27],
        [10_000, 301.211_650_558_652_76],
      ],
      [
        [1_158_691.416_562_977_7, 129.603_510_582_840_27],
        [10_000_000, 129.603_510_582_840_27],
      ],
    ],
    line_percentile_50: [
      [
        [1_158_691.416_562_977_7, 129.603_510_582_840_27],
        [460.983_421_919_825_5, 519.999_999_999_999_8],
      ],
      [
        [1_158_691.416_562_977_7, 129.603_510_582_840_27],
        [10_000_000, 129.603_510_582_840_27],
      ],
    ],
    line_percentile_90: [
      [
        [1_158_691.416_562_977_7, 129.603_510_582_840_27],
        [10_000, 301.211_650_558_652_76],
      ],
      [
        [1_158_691.416_562_977_7, 129.603_510_582_840_27],
        [10_000_000, 129.603_510_582_840_27],
      ],
    ],
    line_percentile_99: [
      [
        [1_158_691.416_562_977_7, 129.603_510_582_840_27],
        [10_000, 301.211_650_558_652_76],
      ],
      [
        [1_158_691.416_562_977_7, 129.603_510_582_840_27],
        [10_000_000, 129.603_510_582_840_27],
      ],
    ],
    percentile_1: 2022,
    percentile_10: 2022,
    percentile_50: 2022,
    percentile_90: 2022,
    percentile_99: 2022,
    plot1: [
      [10_000, 400],
      [2022, 400],
      [2022, 103.682_808_466_272_18],
      [2022, 400],
    ],
    plot2: [
      [10_000, 301.211_650_558_652_65],
      [460.983_421_919_825_5, 520],
      [1_158_691.416_562_977_7, 129.603_510_582_840_22],
      [10_000_000, 129.603_510_582_840_22],
    ],
    slope: 5.635_329_994_802_062,
    stress_amplitude: 400,
    x0: 460.983_421_919_825_5,
    xlim0: 10_000,
    xlim1: 10_000_000,
    y0: 520,
  },
};

export const mockedPredictionResult: PredictionResult = {
  woehler: {
    snCurve: {
      '0': {
        x: 460.983_421_919_825_5,
        y: 520,
      },
      '1': {
        x: 1_158_691.416_562_977_7,
        y: 129.603_510_582_840_22,
      },
      '2': {
        x: 10_000_000,
        y: 129.603_510_582_840_22,
      },
    },
    snCurveLow: {
      '0': {
        x: 460.983_421_919_825_5,
        y: 520,
      },
      '1': {
        x: 1_158_691.416_562_977_7,
        y: 129.603_510_582_840_22,
      },
      '2': {
        x: 10_000_000,
        y: 129.603_510_582_840_22,
      },
    },
    snCurveHigh: {
      '0': {
        x: 460.983_421_919_825_5,
        y: 520,
      },
      '1': {
        x: 1_158_691.416_562_977_7,
        y: 129.603_510_582_840_22,
      },
      '2': {
        x: 10_000_000,
        y: 129.603_510_582_840_22,
      },
    },
    appliedStress: {
      '0': {
        x: 2022,
        y: 400,
      },
      '1': {
        x: 2022,
        y: 103.682_808_466_272_18,
      },
      '2': {
        x: 10_000,
        y: 400,
      },
    },
    percentile1: {
      '0': {
        x: 460.983_421_919_825_5,
        y: 519.999_999_999_999_8,
      },
      '1': {
        x: 1_158_691.416_562_977_7,
        y: 129.603_510_582_840_27,
      },
      '2': {
        x: 10_000_000,
        y: 129.603_510_582_840_27,
      },
    },
    percentile10: {
      '0': {
        x: 460.983_421_919_825_5,
        y: 519.999_999_999_999_8,
      },
      '1': {
        x: 1_158_691.416_562_977_7,
        y: 129.603_510_582_840_27,
      },
      '2': {
        x: 10_000_000,
        y: 129.603_510_582_840_27,
      },
    },
    percentile90: {
      '0': {
        x: 460.983_421_919_825_5,
        y: 519.999_999_999_999_8,
      },
      '1': {
        x: 1_158_691.416_562_977_7,
        y: 129.603_510_582_840_27,
      },
      '2': {
        x: 10_000_000,
        y: 129.603_510_582_840_27,
      },
    },
    percentile99: {
      '0': {
        x: 460.983_421_919_825_5,
        y: 519.999_999_999_999_8,
      },
      '1': {
        x: 1_158_691.416_562_977_7,
        y: 129.603_510_582_840_27,
      },
      '2': {
        x: 10_000_000,
        y: 129.603_510_582_840_27,
      },
    },
  },
  haigh: {
    snCurve: {
      '0': {
        x: 0,
        y: 129.603_510_582_840_22,
      },
      '1': {
        x: 127.486_713_376_344_78,
        y: 127.486_713_376_344_78,
      },
    },
    appliedStress: {
      '0': {
        x: 0,
        y: 0,
      },
      '1': {
        x: 229.603_510_582_840_22,
        y: 229.603_510_582_840_22,
      },
    },
  },
  kpi: {
    fatigue: {
      '0': 127.486_713_376_344_78,
      '1': 129.603_510_582_840_22,
    },
    slope: 5.635_329_994_802_062,
  },
};

export const mockedStatisticalResult = {
  woehler: {
    analytical: {
      sa_fkm: 267.299_999_999_999_95,
      sa_murakami: 357.891_061_943_249_67,
    },
    statistical_sn_curve: {
      percentile_50: {
        '0': { x: 10_000, y: 255.307_159_187_472_5 },
        '1': { x: 329_718.177_015_445_14, y: 122.464_994_125_332_68 },
        '2': { x: 10_000_000, y: 122.464_994_125_332_68 },
      },
      percentile_10: {
        '0': { x: 10_000, y: 342.940_004_948_719_3 },
        '1': { x: 329_718.177_015_445_14, y: 164.500_462_208_140_25 },
        '2': { x: 10_000_000, y: 164.500_462_208_140_25 },
      },
      percentile_90: {
        '0': { x: 10_000, y: 190.067_488_749_596_95 },
        '1': { x: 329_718.177_015_445_14, y: 91.171_019_125_413_8 },
        '2': { x: 10_000_000, y: 91.171_019_125_413_8 },
      },
    },
  },
  haigh: {
    analytical: {
      r0_fkm: 267.299_999_999_999_95,
      r1_fkm: 241.267_262_388_302_19,
      r0_murakami: 357.891_061_943_249_67,
      r1_murakami: 302.203_530_305_919_74,
    },
    statistical: { r0: 108.093_165_672_033_12, r1: 122.464_994_125_332_68 },
  },
};

export const mockedPredictionRequestWithKpi: PredictionState = {
  predictionRequest: {
    prediction: 0,
    mpa: 400,
    v90: 0,
    hv: 180,
    hv_lower: 180,
    hv_upper: 180,
    rrelation: -1,
    burdeningType: 0,
    model: 5,
    spreading: 0,
    rArea: 5,
    es: 0,
    rz: 0,
    hv_core: 500,
    a90: 100,
    gradient: 1,
    multiaxiality: 0,
  },
  statisticalRequest: {
    rz: 0,
    es: 0,
    hardness: 180,
    r: -1,
    rArea: 5,
    v90: 1,
    loadingType: 0,
  },
  statisticalResult: {
    woehler: {
      analytical: {
        sa_fkm: 67.299_999_999_999_95,
        sa_murakami: 293.891_061_943_249_67,
      },
      statistical_sn_curve: {
        percentile_50: {
          0: { x: 10_000, y: 255.307_159_187_472_5 },
          1: { x: 329_718.177_015_445_14, y: 122.464_994_125_332_68 },
          2: { x: 10_000_000, y: 122.464_994_125_332_68 },
        },
        percentile_10: {
          0: { x: 10_000, y: 342.940_004_948_719_3 },
          1: { x: 329_718.177_015_445_14, y: 164.500_462_208_140_25 },
          2: { x: 10_000_000, y: 164.500_462_208_140_25 },
        },
        percentile_90: {
          0: { x: 10_000, y: 190.067_488_749_596_95 },
          1: { x: 329_718.177_015_445_14, y: 91.171_019_125_413_8 },
          2: { x: 10_000_000, y: 91.171_019_125_413_8 },
        },
      },
    },
    haigh: {
      analytical: {
        r0_fkm: 267.299_999_999_999_95,
        r1_fkm: 241.267_262_388_302_19,
        r0_murakami: 357.891_061_943_249_67,
        r1_murakami: 302.203_530_305_919_74,
      },
      statistical: { r0: 108.093_165_672_033_12, r1: 122.464_994_125_332_68 },
    },
  },
  predictionResult: {
    haigh: undefined,
    woehler: undefined,
    kpi: {
      fatigue: {
        0: 10,
        1: 20,
      },
      slope: 120,
    },
  },
};

export const mockedPredictionRequestWithLimits: LTPState = {
  input: {
    ...initialState,
    display: {
      ...initialState.display,
      showFKM: true,
      showMurakami: true,
      showStatistical: true,
    },
  },
  prediction: {
    ...mockedPredictionRequestWithKpi,
    predictionResult: {
      ...mockedPredictionResult,
    },
  },
};

export const mockedPredictionResultGraphData: EChartsOption = {
  dataset: {
    source: [
      {
        x: 10_000,
        y7: 169.049_956_840_594_63,
      },
      {
        x: 1_000_000,
        y7: 67.299_999_999_999_95,
      },
      {
        x: 10_000_000,
        y7: 67.299_999_999_999_95,
      },
      {
        x: 10_000,
        y8: 738.220_970_837_190_4,
      },
      {
        x: 1_000_000,
        y8: 293.891_061_943_249_67,
      },
      {
        x: 10_000_000,
        y8: 293.891_061_943_249_67,
      },
      {
        x: 10_000,
        y9: 255.307_159_187_472_5,
      },
      {
        x: 1_000_000,
        // x: 329718.17701544514,
        y9: 122.464_994_125_332_68,
      },
      {
        x: 10_000_000,
        y9: 122.464_994_125_332_68,
      },
      {
        x: 10_000_000,
        yHigh: 129.603_510_582_840_22,
        yLow: 129.603_510_582_840_22,
      },
      {
        x: 1_158_691.416_562_977_7,
        yHigh: 129.603_510_582_840_22,
        yLow: 129.603_510_582_840_22,
      },
      {
        x: 10_000,
        y2: 301.211_650_558_652_65,
      },
      {
        x: 1_158_691.416_562_977_7,
        y2: 129.603_510_582_840_22,
      },
      {
        x: 10_000_000,
        y2: 129.603_510_582_840_22,
      },
      {
        x: 10_000,
        y3: 301.211_650_558_652_54,
      },
      {
        x: 1_158_691.416_562_977_7,
        y3: 129.603_510_582_840_27,
      },
      {
        x: 10_000_000,
        y3: 129.603_510_582_840_27,
      },
      {
        x: 10_000,
        y5: 301.211_650_558_652_54,
      },
      {
        x: 1_158_691.416_562_977_7,
        y5: 129.603_510_582_840_27,
      },
      {
        x: 10_000_000,
        y5: 129.603_510_582_840_27,
      },
      {
        x: 10_000,
        y6: 301.211_650_558_652_54,
      },
      {
        x: 1_158_691.416_562_977_7,
        y6: 129.603_510_582_840_27,
      },
      {
        x: 10_000_000,
        y6: 129.603_510_582_840_27,
      },
      {
        x: 10_000,
        y4: 301.211_650_558_652_54,
      },
      {
        x: 1_158_691.416_562_977_7,
        y4: 129.603_510_582_840_27,
      },
      {
        x: 10_000_000,
        y4: 129.603_510_582_840_27,
      },
    ],
  },
  xAxis: {
    max: 10_000_000,
    min: 10_000,
  },
  yAxis: {
    max: 812.043_067_920_909_5,
    min: 60.569_999_999_999_96,
  },
  // eslint-disable-next-line max-lines
};
