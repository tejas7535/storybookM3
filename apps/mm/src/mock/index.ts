import { BearinxDialogDescription } from '@caeonline/dynamic-forms';

import { environment } from '../environments/environment';
import * as dialog from './dialog.json';

const envDialogString = JSON.stringify(dialog).replace(
  /OPTIONS_URL/g,
  environment.baseUrl
);

const envDialog = JSON.parse(envDialogString);

export const DIALOG = (envDialog as any).default as BearinxDialogDescription;
