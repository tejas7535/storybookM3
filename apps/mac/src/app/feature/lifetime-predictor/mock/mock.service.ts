import { Injectable } from '@angular/core';

import { Observable, of } from 'rxjs';

import { PredictionRequest } from '../models';
import { mockedPredictionResult } from './mock.constants';

@Injectable({
  providedIn: 'root',
})
export class MockService {
  postPrediction(
    _predictionRequest: PredictionRequest,
    _otherValue: any
  ): Observable<any> {
    return of(mockedPredictionResult);
  }
}
