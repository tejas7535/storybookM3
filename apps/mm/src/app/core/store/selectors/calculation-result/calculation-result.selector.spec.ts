import {
  CalculationResultState,
  MountingTools,
} from '@mm/core/store/models/calculation-result-state.model';
import { APP_STATE_MOCK } from '@mm/testing/mocks/store/app-state.mock';

import {
  getCalculationInputs,
  getCalculationMessages,
  getEndPositions,
  getMountingRecommendations,
  getMountingTools,
  getReportSelectionTypes,
  getStartPositions,
  getVersions,
  hasMountingTools,
  isResultAvailable,
} from './calculation-result.selector';

describe('CalculationResultSelector', () => {
  const state: CalculationResultState = {
    ...APP_STATE_MOCK.calculationResult,
  };

  describe('getCalculationInputs', () => {
    describe('when inputs are not defined', () => {
      it('should return undefined', () => {
        const result = getCalculationInputs.projector(state);

        expect(result).toBeUndefined();
      });
    });

    describe('when inputs are defined', () => {
      it('should return inputs', () => {
        expect(
          getCalculationInputs({
            calculationResult: {
              result: {
                inputs: [
                  {
                    hasNestedStructure: true,
                    title: 'some title',
                  },
                ],
              },
            },
          })
        ).toMatchSnapshot();
      });
    });
  });

  describe('getMountingRecommendations', () => {
    describe('when mounting recommendations are not defined', () => {
      it('should return empty array', () => {
        const result = getMountingRecommendations.projector(state);
        expect(result).toEqual([]);
      });
    });

    describe('when mounting recommendations are defined', () => {
      it('should return mounting recommendations', () => {
        expect(
          getMountingRecommendations({
            calculationResult: {
              result: {
                mountingRecommendations: [
                  'first recommendation',
                  'second recommendation',
                ],
              },
            },
          })
        ).toMatchSnapshot();
      });
    });
  });

  describe('getMountingTools', () => {
    describe('when mounting tools are not defined', () => {
      it('should return empty object', () => {
        const result = getMountingTools.projector(state);
        expect(result).toBeUndefined();
      });
    });

    describe('when mounting tools are defined', () => {
      it('should return mounting tools', () => {
        expect(
          getMountingTools({
            calculationResult: {
              result: {
                mountingTools: {
                  additionalTools: [],
                  hydraulicNut: [],
                  pumps: {
                    title: '',
                    items: [],
                  },
                  locknut: [],
                  sleeveConnectors: [],
                } as MountingTools,
              },
            },
          })
        ).toMatchSnapshot();
      });
    });
  });

  describe('getCalculationMessages', () => {
    describe('when messages are not defined', () => {
      it('should return default messages', () => {
        const result = getCalculationMessages.projector(state);
        expect(result).toEqual({
          errors: [],
          notes: [],
          warnings: [],
        });
      });
    });

    describe('when messages are defined', () => {
      it('should return messages', () => {
        expect(
          getCalculationMessages({
            calculationResult: {
              result: {
                reportMessages: {
                  notes: ['some note'],
                  warnings: [],
                  errors: [],
                },
              },
            },
          })
        ).toMatchSnapshot();
      });
    });
  });

  describe('isResultAvailable', () => {
    describe('when result is not available', () => {
      it('should return false', () => {
        const result = isResultAvailable.projector(state);
        expect(result).toBe(false);
      });
    });

    describe('when result is available', () => {
      it('should return true', () => {
        expect(
          isResultAvailable({
            calculationResult: {
              result: {},
            },
          })
        ).toBe(true);
      });
    });

    describe('when old result is available but is loading a new result', () => {
      it('should return false', () => {
        expect(
          isResultAvailable({
            calculationResult: {
              result: {},
              isLoading: true,
            },
          })
        ).toBe(false);
      });
    });

    describe('hasMountingTools', () => {
      describe('when mounting tools are not available', () => {
        it('should return false', () => {
          expect(
            hasMountingTools({
              calculationResult: {
                result: {},
              },
            })
          ).toBe(false);
        });
      });

      describe('when mounting tools are available', () => {
        const mountingTools: MountingTools = {
          additionalTools: [],
          hydraulicNut: [],
          pumps: {
            title: '',
            items: [],
          },
          locknut: [],
          sleeveConnectors: [],
        } as MountingTools;

        it('should return true when hydraulic nut provided', () => {
          expect(
            hasMountingTools({
              calculationResult: {
                result: {
                  mountingTools: {
                    ...mountingTools,
                    hydraulicNut: [
                      {
                        value: 'some value',
                        unit: 'unit',
                        designation: 'designation value',
                      },
                    ],
                  },
                },
              },
            })
          ).toBe(true);
        });

        it('should return true when additional tools provided', () => {
          expect(
            hasMountingTools({
              calculationResult: {
                result: {
                  mountingTools: {
                    ...mountingTools,
                    additionalTools: [
                      {
                        value: 'some value',
                        unit: 'unit',
                        designation: 'designation value',
                      },
                    ],
                  },
                },
              },
            })
          ).toBe(true);
        });

        it('should return true when pumps provided', () => {
          expect(
            hasMountingTools({
              calculationResult: {
                result: {
                  mountingTools: {
                    ...mountingTools,
                    pumps: {
                      title: 'pumps title',
                      items: [
                        {
                          field: 'some field',
                          value: 'some value',
                          isRecommended: true,
                        },
                      ],
                    },
                  },
                },
              },
            })
          ).toBe(true);
        });

        it('should return true when locknut provided', () => {
          expect(
            hasMountingTools({
              calculationResult: {
                result: {
                  mountingTools: {
                    ...mountingTools,
                    locknut: [
                      {
                        value: 'some value',
                        unit: 'unit',
                        designation: 'designation value',
                      },
                    ],
                  },
                },
              },
            })
          ).toBe(true);
        });
      });
    });

    describe('getReportSelectionTypes', () => {
      describe('when no reports types are available', () => {
        it('should return empty list', () => {
          expect(
            getReportSelectionTypes({
              calculationResult: {
                result: {},
              },
            })
          ).toEqual([]);
        });
      });

      describe('when reports types are available', () => {
        it('should return types list', () => {
          expect(
            getReportSelectionTypes({
              calculationResult: {
                result: {
                  reportMessages: {
                    notes: ['some note'],
                    warnings: [],
                    errors: [],
                  },
                  mountingTools: {
                    additionalTools: [
                      {
                        value: 'some value',
                        unit: 'unit',
                        designation: 'designation value',
                      },
                    ],
                    hydraulicNut: [],
                    pumps: {
                      title: '',
                      items: [],
                    },
                    locknut: [],
                    sleeveConnectors: [],
                  } as MountingTools,
                  mountingRecommendations: [
                    'first recommendation',
                    'second recommendation',
                  ],
                  inputs: [
                    {
                      hasNestedStructure: true,
                      title: 'some title',
                    },
                  ],
                },
              },
            })
          ).toMatchSnapshot();
        });
      });
    });
  });
  describe('getVersions', () => {
    const mockState = {
      calculationResult: {},
    };
    it('should return undefined if there are no versions', () => {
      expect(
        getVersions({
          ...mockState,
          calculationResult: {
            ...mockState.calculationResult,
            versions: undefined,
          },
        })
      ).toBeUndefined();
    });

    it('should return undefined if the versions object is empty', () => {
      expect(
        getVersions({
          ...mockState,
          calculationResult: {
            ...mockState.calculationResult,
            versions: {},
          },
        })
      ).toBeUndefined();
    });

    it('should return the versions as a string', () => {
      expect(
        getVersions({
          ...mockState,
          calculationResult: {
            ...mockState.calculationResult,
            versions: {
              '1': 'v1',
              '2': 'v2',
            },
          },
        })
      ).toEqual('1 v1 / 2 v2');
    });
  });
  describe('getStartPositions', () => {
    describe('when start positions are not defined', () => {
      it('should return empty array', () => {
        const result = getStartPositions.projector(state);
        expect(result).toEqual([]);
      });
    });

    describe('when start positions are defined', () => {
      it('should return start positions', () => {
        expect(
          getStartPositions({
            calculationResult: {
              result: {
                startPositions: [
                  { abbreviation: 'pos1', value: '10 mm', title: 'Position 1' },
                  { abbreviation: 'pos2', value: '20 mm', title: 'Position 2' },
                ],
              },
            },
          })
        ).toEqual([
          { abbreviation: 'pos1', value: '10 mm', title: 'Position 1' },
          { abbreviation: 'pos2', value: '20 mm', title: 'Position 2' },
        ]);
      });
    });
  });

  describe('getEndPositions', () => {
    describe('when end positions are not defined', () => {
      it('should return empty array', () => {
        const result = getEndPositions.projector(state);
        expect(result).toEqual([]);
      });
    });

    describe('when end positions are defined', () => {
      it('should return end positions', () => {
        expect(
          getEndPositions({
            calculationResult: {
              result: {
                endPositions: [
                  {
                    abbreviation: 'pos1',
                    value: '15 mm',
                    title: 'End Position 1',
                  },
                  {
                    abbreviation: 'pos2',
                    value: '25 mm',
                    title: 'End Position 2',
                  },
                ],
              },
            },
          })
        ).toEqual([
          { abbreviation: 'pos1', value: '15 mm', title: 'End Position 1' },
          { abbreviation: 'pos2', value: '25 mm', title: 'End Position 2' },
        ]);
      });
    });

    describe('when result is null', () => {
      it('should return empty array', () => {
        expect(
          getEndPositions({
            calculationResult: {
              // eslint-disable-next-line unicorn/no-null
              result: null,
            },
          })
        ).toEqual([]);
      });
    });
    describe('getReportSelectionTypes', () => {
      describe('when no inputs, mounting tools, recommendations, messages, start positions, or end positions are available', () => {
        it('should return an empty array', () => {
          const result = getReportSelectionTypes.projector(
            undefined,
            false,
            [],
            { errors: [], warnings: [], notes: [] },
            [],
            [],
            [],
            []
          );
          expect(result).toEqual([]);
        });
      });

      describe('when start positions are available', () => {
        it('should include "startPosition" in the result', () => {
          const result = getReportSelectionTypes.projector(
            undefined,
            false,
            [],
            { errors: [], warnings: [], notes: [] },
            [
              {
                abbreviation: 'pos1',
                value: '10',
                designation: 'Position 1',
                unit: 'mm',
              },
            ],
            [],
            [],
            []
          );
          expect(result).toEqual([{ name: 'startPosition' }]);
        });
      });

      describe('when end positions are available', () => {
        it('should include "endPosition" in the result', () => {
          const result = getReportSelectionTypes.projector(
            undefined,
            false,
            [],
            { errors: [], warnings: [], notes: [] },
            [],
            [
              {
                abbreviation: 'pos1',
                value: '15',
                designation: 'End Position 1',
                unit: 'mm',
              },
            ],
            [],
            []
          );
          expect(result).toEqual([{ name: 'endPosition' }]);
        });
      });

      describe('when inputs are available', () => {
        it('should include "reportInputs" in the result', () => {
          const result = getReportSelectionTypes.projector(
            [{ hasNestedStructure: true, title: 'some title' }],
            false,
            [],
            { errors: [], warnings: [], notes: [] },
            [],
            [],
            [],
            []
          );
          expect(result).toEqual([{ name: 'reportInputs' }]);
        });
      });

      describe('when mounting tools are available', () => {
        it('should include "mountingToolsAndUtilities" in the result', () => {
          const result = getReportSelectionTypes.projector(
            undefined,
            true,
            [],
            { errors: [], warnings: [], notes: [] },
            [],
            [],
            [],
            []
          );
          expect(result).toEqual([{ name: 'mountingToolsAndUtilities' }]);
        });
      });

      describe('when mounting recommendations are available', () => {
        it('should include "mountingInstructions" in the result', () => {
          const result = getReportSelectionTypes.projector(
            undefined,
            false,
            ['first recommendation', 'second recommendation'],
            { errors: [], warnings: [], notes: [] },
            [],
            [],
            [],
            []
          );
          expect(result).toEqual([{ name: 'mountingInstructions' }]);
        });
      });

      describe('when messages are available', () => {
        it('should include "reportMessages" in the result if there are errors', () => {
          const result = getReportSelectionTypes.projector(
            undefined,
            false,
            [],
            { errors: ['some error'], warnings: [], notes: [] },
            [],
            [],
            [],
            []
          );
          expect(result).toEqual([{ name: 'reportMessages' }]);
        });

        it('should include "reportMessages" in the result if there are warnings', () => {
          const result = getReportSelectionTypes.projector(
            undefined,
            false,
            [],
            { errors: [], warnings: ['some warning'], notes: [] },
            [],
            [],
            [],
            []
          );
          expect(result).toEqual([{ name: 'reportMessages' }]);
        });

        it('should include "reportMessages" in the result if there are notes', () => {
          const result = getReportSelectionTypes.projector(
            undefined,
            false,
            [],
            { errors: [], warnings: [], notes: ['some note'] },
            [],
            [],
            [],
            []
          );
          expect(result).toEqual([{ name: 'reportMessages' }]);
        });
      });

      describe('when multiple conditions are met', () => {
        it('should include all applicable report selection types', () => {
          const result = getReportSelectionTypes.projector(
            [{ hasNestedStructure: true, title: 'some title' }],
            true,
            ['first recommendation'],
            { errors: ['some error'], warnings: [], notes: [] },
            [
              {
                abbreviation: 'pos1',
                value: '10',
                designation: 'Position 1',
                unit: 'mm',
              },
            ],
            [
              {
                abbreviation: 'pos2',
                value: '15',
                designation: 'End Position 1',
                unit: 'mm',
              },
            ],
            [
              {
                designation: 'some designation',
                value: 'some value',
                unit: 'some unit',
              },
            ],
            [
              {
                designation: 'some designation',
                value: 'some value',
                unit: 'some unit',
              },
            ]
          );
          expect(result).toMatchSnapshot();
        });
      });
    });
  });
});
