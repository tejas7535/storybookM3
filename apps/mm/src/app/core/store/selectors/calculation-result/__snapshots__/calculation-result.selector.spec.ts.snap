// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculationResultSelectors getCalculationInputs when inputs are defined should return inputs 1`] = `
Array [
  Object {
    "hasNestedStructure": true,
    "title": "some title",
  },
]
`;

exports[`CalculationResultSelectors getCalculationMessages when messages are defined should return messages 1`] = `
Object {
  "errors": Array [],
  "notes": Array [
    "some note",
  ],
  "warnings": Array [],
}
`;

exports[`CalculationResultSelectors getEndPositions getReportSelectionTypes when multiple conditions are met should include all applicable report selection types 1`] = `
Array [
  Object {
    "name": "reportInputs",
  },
  Object {
    "name": "startPosition",
  },
  Object {
    "name": "endPosition",
  },
  Object {
    "name": "radialClearance",
  },
  Object {
    "name": "clearanceClasses",
  },
  Object {
    "name": "mountingToolsAndUtilities",
  },
  Object {
    "name": "mountingInstructions",
  },
  Object {
    "name": "reportMessages",
  },
]
`;

exports[`CalculationResultSelectors getMountingRecommendations when mounting recommendations are defined should return mounting recommendations 1`] = `
Array [
  "first recommendation",
  "second recommendation",
]
`;

exports[`CalculationResultSelectors getMountingTools when mounting tools are defined should return mounting tools 1`] = `
Object {
  "additionalTools": Array [],
  "hydraulicNut": Array [],
  "locknut": Array [],
  "pumps": Object {
    "items": Array [],
    "title": "",
  },
  "sleeveConnectors": Array [],
}
`;

exports[`CalculationResultSelectors isResultAvailable getReportSelectionTypes when reports types are available should return types list 1`] = `
Array [
  Object {
    "name": "reportInputs",
  },
  Object {
    "name": "mountingToolsAndUtilities",
  },
  Object {
    "name": "mountingInstructions",
  },
  Object {
    "name": "reportMessages",
  },
]
`;
